### ISSUE TITLE
Cannot create image with bud command

### ISSUE THREAD

**Issue by q384566678** [projectatomic/buildah#539]
The following error occurs when I execute the following command to create an image:

```
➜  /data sudo buildah bud -f Dockerfile 
STEP 1: FROM alpine
error building: error creating build container: no such image "alpine" in registry: image not known
```
The environment is as follows:
```
➜  /data sudo buildah version
Version:       0.15
Go Version:    go1.10
Image Spec:    1.0.0
Runtime Spec:  1.0.0
Git Commit:    77c5300
Built:         Mon Mar 26 13:46:37 2018
OS/Arch:       linux/amd64
```
But I can use the docker command to build.
What solution do you have? 
Thank you!

**Comment by ripcurld** [comment]
hi @q384566678 can you give information such as:
 * linux distribution
 * configurations (`/etc/containers/{policy.json,registries.conf,storage.conf,/registries/*}`)

I tried to reproduce the error but unsuccessfully:
```
(0) vagrant@buildah $ sudo ./buildah bud -f ~/temp/df2/Dockerfile 
STEP 1: FROM alpine
STEP 2: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]@2cbc52a3f93e6d60e930d4f6830b5dadf19ff8f4101e8ed2a8508b20e85826b1
Getting image source signatures
Skipping fetch of repeat blob sha256:cd7100a72410606589a54b932cabd804a17f9ae5b42a1882bd56d263e02b6215
Skipping fetch of repeat blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
Copying config sha256:f8c14cf88e6586cb479a016ddfc5c108dc516912f92783a54a7553a8fb6cc121
 639 B / 639 B [============================================================] 0s
Writing manifest to image destination
Storing signatures
```

**Comment by TomSweeneyRedHat** [comment]
@q384566678 Thanks for the report.  +1 on what @ripcurld asked for.  This looks very much like a registry resolution problem.  My guess is something is off in your /etc/containers/registries.conf file.  If you've a chance, change the line in your Docker file to:

```
FROM docker://docker.io/alpine 
```
If that works, then we are on the right track.  Thanks!

**Comment by q384566678** [comment]
Thanks for your suggestions. I don't find `/etc/containers` folder, I try to set up a `containers` folder under `/etc`, and put the `buildah/tests/registries.conf` copied to the `containers` below, but still not success.

This is my Linux distribution:
```
➜  ~ uname -a
Linux zz 4.15.10-1-ARCH #1 SMP PREEMPT Thu Mar 15 12:24:34 UTC 2018 x86_64 GNU/Linux
```

**Comment by rhatdan** [comment]
https://github.com/q384566678 did you try Tom's suggestion?

**Comment by q384566678** [comment]
```
➜  /data sudo buildah bud -f Dockerfile 
STEP 1: FROM docker://docker.io/alpine
error building: error creating build container: no such image "docker://docker.io/alpine" in registry: image not known
```

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Do we have any information about the location of registries.conf in buildah, or is this all buried in containers/image?  We could improve the message when it returns can'f find the image to check if the registries.conf file exist, and report that as well.  It would be better if we knew that the user did not fully specify the image location.

@baude WDYT?

**Comment by ripcurld** [comment]
@q384566678 `uname -a` doesn't give information about your distribution. Let me rephrase the question: are you using CentOS, Fedora, RHEL, Ubuntu, Debian or Arch?
In addition, can you describe how you install buildah?

If it helps these are my configs:
```
vagrant@~ $ cat /etc/containers/registries.conf 
[registries.search]
registries = ['docker.io', 'registry.fedoraproject.org', 'registry.access.redhat.com']

[registries.insecure]
registries = []

[registries.block]
registries = []

vagrant@~ $ cat /etc/containers/storage.conf
[storage]
driver = "overlay"
runroot = "/var/run/containers/storage"
graphroot = "/var/lib/containers/storage"

[storage.options]
additionalimagestores = []
size = ""
override_kernel_check = "true"

vagrant@~ $ cat /etc/containers/policy.json                                                    
{                                                                                                  
    "default": [
        {
            "type": "insecureAcceptAnything"
        }
    ],
    "transports":
        {
            "docker-daemon":
                {
                    "": [{"type":"insecureAcceptAnything"}]
                }
        }
}

vagrant@~ $ cat /etc/containers/registries.d/default.yaml                                      
default-docker:
  sigstore-staging: file:///var/lib/atomic/sigstore
```

My distribution is `Fedora 27`:
```
vagrant@~ $ cat /etc/fedora-release                                                            
Fedora release 27 (Twenty Seven)  
```

**Comment by TomSweeneyRedHat** [comment]
@q384566678 you should get /etc/containers when you install the rpm or if you install the dependencies.  Can you tell me how you installed Buildah?  Did you use an RPM or did you clone/build from GitHub?  Assuming you're on Fedora, can you try doing this if you did not install from an RPM?  You probably want to delete your /etc/containers directory and it's contents before hand.

```
 dnf -y install \
    make \
    golang \
    bats \
    btrfs-progs-devel \
    device-mapper-devel \
    glib2-devel \
    gpgme-devel \
    libassuan-devel \
    libseccomp-devel \
    ostree-devel \
    git \
    bzip2 \
    go-md2man \
    runc \
    skopeo-containers
```

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I'm not seeing much info about registries.conf in docs after a brief scan.  I think it's all in container/images.  I'll create an issue to document that.  Might be a good tutorial/mini blog.

**Comment by ripcurld** [comment]
I'm closing this as it's not a **bug** and related to installing `buildah`.
Although I am closing this, feel free to continue your conversation and if needed open it again.

Note: @TomSweeneyRedHat opened a follow up issue here: https://github.com/projectatomic/buildah/issues/542

Thanks.

**Comment by q384566678** [comment]
@ripcurld @TomSweeneyRedHat @rhatdan Thank you very much, I have already solved this problem. This is because I did not install `skopeo-containers`. I just cloned `buildah` from github. My linux distribution was arch. There were some things that were not found and installed. Then I directly executed `make all`.

**Comment by ripcurld** [comment]
@q384566678 that's really interesting. I have never seen someone using `buildah` on `arch`.
Feel free to share with us your journey running `buildah`/`podman`.
Thanks.

**Comment by rhatdan** [comment]
Opened up a PR to begin discussing how we could make this easier for someone playing with buildah/podman to diagnose.
